test/test_csvs.py

```diff
@@ -0,0 +1,289 @@
+# OOIPLACEHOLDER
+#
+# Copyright 2018 Raytheon Co.
+
+import os
+import unittest
+from nose.plugins.attrib import attr
+
+@attr('UNIT')
+class GeneralUnitTest(unittest.TestCase):
+    
+    def _getPath(self, filename):
+        return os.path.join(os.path.dirname(__file__), filename)
+
+    def _getParameters(self):
+        parameters = {} # dictionary: key=name; value=data type number
+        with open(self._getPath('parameter.txt'), mode='rt') as f:
+            for line in f:
+                parts = line.strip().split(':')
+                parameters[parts[0]] = parts[1]
+        return parameters
+
+    def _getRefDesignators(self):
+        refdes = set()
+        with open(self._getPath('refdes.txt'), mode='rt') as f:
+            for line in f:
+                refdes.add(line.strip())
+        return refdes
+
+    def _getTypes(self):
+        types = {} # dictionary: key=number; value=data type
+        with open(self._getPath('type.txt'), mode='rt') as f:
+            for line in f:
+                parts = line.strip().split(':')
+                types[parts[0]] = parts[1]
+        return types
+
+    def _getUnits(self):
+        units = set()
+        units.add('') # empty value is valid
+        with open(self._getPath('unit.txt'), mode='rt') as f:
+            for line in f:
+                units.add(line.strip())
+        return units
+
+    def _isEmptyOrNumeric(self, value):
+        # if it's empty it's ok
+        if len(value) == 0: return True
+
+        # remove leading minus sign if it exists
+        if value[:1] == '-': value = value[1:]
+
+        # if there's only one decimal point remove it
+        if value.count('.') == 1:
+            value = value[:value.index('.')] + value[value.index('.')+1:]
+
+        return value.isdigit()
+
+    def _getValidationCollection(self):
+        units = self._getUnits()
+        types = self._getTypes()
+        parameters = self._getParameters()
+        refdes = self._getRefDesignators()
+
+        return {'ReferenceDesignator' : refdes, \
+                'ParameterID_R'       : parameters, \
+                'ParameterID_T'       : parameters, \
+                '_Units'              : units, \
+                '_DataLevel'          : ('','L0','L1','L2') \
+               }
+
+    def _getValidationPositions(self):
+        return {'_Array ID'    : (0,8), \
+                '_Platform ID' : (9,14), \
+                '_Instrument'  : (18,27)
+               }
+
+    def _validateLine(self,header,fldsToValidate,line):
```

Comment:
Please use PyCharm or another editor that enforces PEP-8 coding standards.

Reply:
I'll download PyCharm and use it.